Jenkinsfile

```diff
@@ -15,7 +15,17 @@ pipeline {
 
     stage('Test') {
       steps {
-        sh 'JAVA_OPTS= ./gradlew clean check'
+        sh 'printenv | sort'
+        sh 'java -version' // print the java version being used, to aid in debugging build issues
+
+        // if the build breaks, and it's necessary to temporarily pin to a known working gradle
+        // version, can do e.g. --gradle-version=8.4
+        sh 'JAVA_OPTS= ./gradlew wrapper --gradle-version=latest --distribution-type=bin'
+        sh './gradlew --version' // print some basic gradle info, to aid in debugging build issues
+
+        // make extra sure that no stale build artifacts are confounding things -- can revisit if builds are
+        // unpleasantly slow
+        sh 'JAVA_OPTS= JDK_JAVA_OPTIONS= ./gradlew --no-build-cache --no-configuration-cache clean check'
```

Comment:
fwiw, i had this comment in my local working copy, uncommitted, from when i was last working on this (late 2023 iirc):
```
// TODO: observations:
// * blanking JDK_JAVA_OPTIONS doesn't fix the build in CI, or break it locally.
// * on laptop, JDK 11 gets me the "An illegal reflective access operation has occurred" error i see in jenkins
// * also on laptop, build does not fail and tests run _despite_ that reflective access warning
// * so it really seems like a difference in encoding outcome vs test expectation, that doesn't show up on laptop with JDK 11 or jenkins linux box with JDK 21?
```